src/iqm/iqm_client/authentication.py

```diff
@@ -0,0 +1,308 @@
+        password: password for logging into the authentication server
+
+    The parameters can also be read from the environment variables IQM_TOKEN, IQM_TOKENS_FILE,
+    IQM_AUTH_SERVER, IQM_AUTH_USERNAME, IQM_AUTH_PASSWORD. Environment variables can not be
+    mixed with initialisation arguments. All parameters must come from the same source.
+    """
+
+    @staticmethod
+    def time_left_seconds(token: str) -> int:
+        """Check how much time is left until the token expires.
+
+        Returns:
+            Time left on token in seconds.
+        """
+        if not token:
+            return 0
+        parts = token.split('.', 2)
+        if len(parts) != 3:
+            return 0
+        # Add padding to adjust body length to a multiple of 4 chars as required by base64 decoding
+        body = parts[1] + ('=' * (-len(parts[1]) % 4))
+        exp_time = int(json.loads(b64decode(body)).get('exp', '0'))
+        return max(0, exp_time - int(time.time()))
+
+    def __init__(
+        self,
+        token: Optional[str] = None,
+        tokens_file: Optional[str] = None,
+        auth_server_url: Optional[str] = None,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+    ):
+        """
+        Initializes and configures the TokenManager.
+
+        Keyword Args:
+            token (str): Optional long-lived token in plain text format.
+                If ``token`` is given no other user authentication parameters should be given.
+            tokens_file (str): Optional path to a tokens file used for authentication.
+                If ``tokens_file`` is given no other user authentication parameters should be given.
+            auth_server_url (str): Optional base URL of the authentication server.
+                If ``auth_server_url`` is given also ``username`` and ``password`` must be given.
+            username (str): Optional username to log in to authentication server.
+            password (str): Optional password to log in to authentication server.
+
+        Alternatively, the user authentication related keyword arguments can also be given in
+        environment variables ``IQM_TOKEN``, ``IQM_TOKENS_FILE``, ``IQM_AUTH_SERVER``,
+        ``IQM_AUTH_USERNAME`` and ``IQM_AUTH_PASSWORD``. All parameters must be given either
+        as keyword arguments or as environment variables.
+        """
+
+        def _format_names(variable_names: list[str]) -> str:
+            """Format a list of variable names"""
+            return ', '.join(f'"{name}"' for name in variable_names)
+
+        auth_parameters: dict[str, str] = {}
+
+        init_parameters = {
+            'token': token,
+            'tokens_file': tokens_file,
+            'auth_server_url': auth_server_url,
+            'username': username,
+            'password': password,
+        }
+        init_params_given = [key for key, value in init_parameters.items() if value]
```

Comment:
```suggestion
        init_params_given = {key: value for key, value in init_parameters.items() if value}
```
Maybe including the keys would be more informative in the error message?

Reply:
Same for env_vars_given

Reply:
improved the error message